docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+- Static nodes such as `exporter` (`10000`)
+
+
+### User Agent
+
+Libp2p provides user agent mechanism with the [identify](https://github.com/libp2p/specs/tree/master/identify) protocol, 
+which is used to exchange basic information with other peers in the network.
+
+User Agent contains the node version and type, and in addition the operator id which might be reduced in future versions. \
+See detailed format in [forks > user agent](#fork-v0)
+
+
+### Discovery
+
+[discv5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5.md) is a system for finding other participants in a peer-to-peer network, 
+it is used in `SSV.network` to complement discovery.
+
+DiscV5 works on top of UDP, it uses a DHT to store node records (`ENR`) of discovered peers. \
+The discovery process walks randomaly on the nodes in the table that are not connected, 
+filters them by `ENR` entries in order to connect with the most relevant nodes.
+
+As discv5 is standalone (i.e. not depends on libp2p), the communication is encrypted and authenticated using session keys,
+established in ana separate [handshake process](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md#sessions).
+
+**Bootnode** 
+
+A peer that have a public, static ENR to enable new peers to join the network. For the sake of flexibility, 
+bootnode/s ENR values are configurable and can be changed on demand by operators. \
+Bootnode doesn't start a libp2p host for TCP communication, its role ends once a new peer finds existing peers in the network.
+
+#### ENR
+
+[Ethereum Node Records](https://github.com/ethereum/devp2p/blob/master/enr.md) is a format that holds peer information.
+Records contain a signature, sequence (for republishing record) and arbitrary key/value pairs. 
+
+`ENR` structure in `SSV.Network` consists of the following key/value pairs:
+
+| Key         | Value                                                          | Status          |
+|:------------|:---------------------------------------------------------------|:----------------|
+| `id`        | name of identity scheme, e.g. "v4"                             | Done            |
+| `secp256k1` | compressed secp256k1 public key, 33 bytes                      | Done            |
+| `ip`        | IPv4 address, 4 bytes                                          | Done            |
+| `tcp`       | TCP port, big endian integer                                   | Done            |
+| `udp`       | UDP port, big endian integer                                   | Done            |
+| `type`      | node type, integer; 1 (operator), 2 (exporter), 3 (bootnode)   | Done (`v0.1.9`) |
+| `oid`       | operator id, 32 bytes                                          | Done (`v0.1.9`) |
+| `version`   | fork version, integer                                          | -               |
+| `subnets`   | bitlist, 0 for irrelevant and 1 for assigned subnet            | -               |
+
+#### Discovery Alternatives
```

Comment:
no need for alternatives in a spec, let's decided on a specific spec and go with it